### ISSUE TITLE
Cannot input quotation marks or backslashes in mitmproxy console

### ISSUE THREAD

**Issue by astiob** [mitmproxy/mitmproxy#3216]
##### Steps to reproduce the problem:

1. Launch `mitmproxy`.
2. Open the command console by pressing `:` or `f`.
3. Type: `"foo"` or `""` or `'foo'` or `""` or `\\` or `\"` or `\'`.

As soon as the second quotation mark is typed in, both quotation marks disappear. As soon as the character after the backslash is typed in, the preceding backslash disappears.

In particular, this makes it impossible to filter by values that contain quotation marks (as already noted in #1053) or backslashes: you either don’t have the desired characters in the filter, or you have only one quotation mark and it complains that `Command error: No closing quotation`, or you have a lone backslash and it complains that `Command error: No escaped character`.

##### Any other comments? What have you tried so far?

I’ve tried to quote the affected characters by doubling them, preceding them with backslashes or enclosing them in other quotation marks, but as you can see, none of this helps.

Setting the view filter from the command line works, but as soon as you try to edit it interactively afterwards, it gets broken in the same manner.

##### System information

```
Mitmproxy: 4.0.1 
Python:    3.6.5
OpenSSL:   OpenSSL 1.1.0h  27 Mar 2018
Platform:  Linux-4.4.114-42-default-x86_64-with-SuSE-42.3-x86_64
```

The same thing happened on mitmproxy 3.

**Comment by mhils** [comment]
Yes, this is super annoying right now. @kajojify's command work should actually fix this! 😃

**Comment by cortesi** [comment]
Thanks for your report! The proximate issue here has been fixed in master via #3183. We still have work to do with string quotes and escaping - I've opened a separate issue for that at #3220.
